src/node_http2_core.h

```diff
@@ -166,13 +166,30 @@ class Nghttp2Session {
                             int error_code) {}
   virtual ssize_t GetPadding(size_t frameLength,
                              size_t maxFrameLength) { return 0; }
-  virtual void OnTrailers(Nghttp2Stream* stream,
-                          MaybeStackBuffer<nghttp2_nv>* nva) {}
   virtual void OnFreeSession() {}
   virtual void AllocateSend(size_t suggested_size, uv_buf_t* buf) = 0;
 
   virtual bool HasGetPaddingCallback() { return false; }
 
+  class SubmitTrailers {
+   public:
+    void operator ()(nghttp2_nv* trailers, size_t length) const;
```

Comment:
Making this a normal method called `.Send()` or whatever is clearer, IMO.  No risk of confusing method calls with plain function calls.